hazelcast/src/main/java/com/hazelcast/executor/impl/DistributedExecutorService.java

```diff
@@ -298,6 +299,9 @@ private Processor(String name, UUID uuid,
 
         @Override
         public void run() {
+            String currentThreadName = Thread.currentThread().getName();
```

Comment:
The change seems correct, I wonder if we could make it more broad though, ie. for the other Executor services to take advantage of it. For example, ScheduledExecutor & DurableExecutor both use the same core logic, delegating to Cached Executors from `ExecutionServiceImpl` thus, they should have the same issue as IExecutorService.
Could this be logic in the ThreadFactory in there, or too complicated?

Reply:
Thanks @tkountis, I'll check the other executors and try to achieve the same effect for all of them.
Not sure how it is possible to make this change in ThreadFactory, since we still want all the threads to be shared, just have different name when they are used to execute tasks for specific executor.
But I will look and attempt that.

Reply:
By the time we hit `ThreadFactory.createThread`, there is nothing we can extract from the `Runnable` to influence the thread name.
I add the logic in `CachedExecutorServiceDelegate`, right before submitting the `Runnable` instead.